proposals/OpaqueReflect.md

```diff
@@ -0,0 +1,132 @@
+# Specification of opaque reflection
+
+## Goal
+
+This PR addresses #1892. The goal is to permit the reflection of functions which contain symbols that are not already in the logic, through the automatic lifting of the symbols which are not in the logic, using measures.
+
+## Syntax and use
+
+### First, real-world example
+
+```Haskell
+{-@ LIQUID "--reflection"      @-}
+
+module OpaqueRefl04 where
+
+{-@ reflect keepEvens @-}
+keepEvens :: [Int] -> [Int]
+keepEvens = filter even
+```
+
+It will introduce measures for filter and even, such that the result would be equivalent to
+
+```Haskell
+{-@ LIQUID "--reflection"      @-}
+
+module OpaqueRefl06 where
+
+{-@ measure GHC.List.filter :: (a -> Bool) -> [a] -> [a] @-}
+{-@ assume GHC.List.filter :: p:(a -> Bool) -> xs:[a] -> {v : [a] | v == GHC.List.filter p xs && len v <= len xs} @-}
+{-@ measure GHC.Real.even :: Integral a => a -> Bool @-}
+{-@ assume GHC.Real.even :: Integral a => x:a -> {VV : Bool | VV == GHC.Real.even x} @-}
```

Comment:
Do typeclass constraints need to be included in the assumptions?

Reply:
No you don't but is there any reason not to?

Reply:
Only to make them more readable. I don't feel strong about this.